docs/website/contents/for-developers/HardForkCombinatorRollbackMonotonicityProposal.md

```diff
@@ -0,0 +1,285 @@
+
+| DST | CCT | PIL | DSR | ZSE | ZBE | QSR |
+| --- | --- | --- | --- | --- | --- | --- |
+| S   | D   | I   | S 2 | D   | D 1 | I 3 |
+
+- (1) ZBE would be achievable for a prefix of eras when constructing the initial HFC ledger state (in the `ProtocolInfo` passed to the Consensus layer), but otherwise the HFC as a block type does not need to support ZBE at all.
+
+- (2) Esgen points out that perhaps DSR could/should still be Insensitive, and that wouldn't obviously lose too much simplicity.
+
+- (3) Having QSR=Insensitive despite DSR=Sensitive requires an additional mechanism.
+  In this design point, that is anticipated to be constraining these queries to only be answered via an immutable ledger state.
+
+## Disclaimer
+
+There may be other points in the design space worth considering.
+(Or even other dimensions, of course.)
+In particular, DST=Insensitive and PIL=Impossible could be achieved by changing the Cardano ledger's governance to match the existing Cardano HFC logic.
+Specifically, if the ledger did not enact governance actions unless there have actually been k+1 blocks since they were ratified even despite possible Chain Growth violations, then the Cardano HFC would never override the Cardano ledger.
+
+However, Frisby thinks such block counting (and the related "double-stability" in the existing ledger rules) is strictly undue complexity that should be removed instead of propagated even deeper.
+
+---
+
+# DRAFT NUMBER 1
+
+## Introduction
+
+This document motivates a simplification of the HFC, that would also alleviate the "double stability" constraint on the ledger's governance rules.
+
+Specifically, it suggests that the HFC should not be vetoing the ledger's decision to transition to the next era while that decision is not yet immutable.
+This veto can only happen in the presence of a Chain Growth violation, but it's confusing none-the-less and requires extra implementation complexity in the HFC.
+
+The suggested simplification will necessarily change some behaviors, but notably the behaviors that principally motivated the HFC's block counting in the first place could be preserved by only answering time-related queries using the immutable ledger state.
+
+## Though Experiment: Predictions within a Simplified Node
+
+Consider a simplified node.
+
+- Only node-to-node protocols: no queries, etc.
+- Pure Praos: no hard fork combinator, no Transitional Praos, etc.
+
+### Question 1
+
+Which aspects of the protocol and ledger's future behavior must this node be able to predict from a given ledger state and why?
+
+- Assume it's well-connected and running in a healthy net.
+- Assume k > 0.
+- Let C be the current best valid chain in the net.
+- Let D be the node's currently selected chain, and assume it's worse than C.
+- Let L be the ledger state resulting from the youngest block that is on both C and D, aka their intersection.
```

Comment:
Now I realize that I'm confused about the mention of "youngest block", doesn't block at the intersection suffice?

Reply:
Yeah; see our previous convo https://github.com/input-output-hk/ouroboros-consensus/pull/346#discussion_r1333010441